### ISSUE TITLE
roachperf: regression around 2022-04-06

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#82136]
We're seeing a roachperf regression around 2022-04-06 on multiple graphs, e.g.:

* [`kv95/enc=false/nodes=3/cpu=32`](https://roachperf.crdb.dev/?filter=&view=kv95%2Fenc%3Dfalse%2Fnodes%3D3%2Fcpu%3D32&tab=aws)
* [`ycsb/B/nodes=3/cpu=32`](https://roachperf.crdb.dev/?filter=&view=ycsb%2FB%2Fnodes%3D3%2Fcpu%3D32&tab=aws)
* [`ycsb/C/nodes=3/cpu=32`](https://roachperf.crdb.dev/?filter=&view=ycsb%2FC%2Fnodes%3D3%2Fcpu%3D32&tab=aws)
* [`ycsb/D/nodes=3/cpu=32`](https://roachperf.crdb.dev/?filter=&view=ycsb%2FD%2Fnodes%3D3%2Fcpu%3D32&tab=aws)

<img width="748" alt="Screenshot 2022-05-31 at 15 03 58" src="https://user-images.githubusercontent.com/644420/171179860-622061f8-b02a-47c8-acbc-e6381e82e157.png">

**Comment by tbg** [comment]
15bbd782fd Merge #77337 #79462 <-- supposedly bad
76df030db4 Merge #79581
16a2652718 Merge #78255 #79434
095192b848 Merge branch 'cockroachdb:master' into master
74e17666a2 Merge branch 'cockroachdb:master' into master
714c95a51f Merge pull request #1 from taroface/log-reference-5
112a00859f Merge #79535
b038da197e Merge #79465 #79523 #79538 #79541 #79561 #79584
4940bc160f Merge #79577
958df7cb67 Merge #79500
d40c144415 Merge #79543
9bff1de711 Merge #79554
bb27f761a9 Merge #79295
9cffc04af5 Merge #79360
38a3442b43 Merge #79522
e54a36cc36 Merge #77960 #79257 #79455
301a424125 Merge #77284 #79010 #79440
f981d447c7 Merge #79468
d4daa99f6b Merge #74005 #77794 #79503 #79532
f46a354069 Merge #79085 #79407 #79473
cf74ec6acd Merge #78423
3b99ec1979 Merge #75759 #79507
f69b88fc2f Merge #78982
6aff9807c8 Merge #79476
92bf06a8d5 Merge #79475
72b71b6920 Merge #77649 #78272 #78448 #79395 #79397
1e48e08fe6 Merge #79350
df37478477 Merge #79365
fc97617d79 Merge #79023
b84398d95a Merge #77742 #79260 #79324 #79389 #79454
6cf835c71f Merge #77521 #79449
985344a22b Merge #76312 #78652 #79022 #79157 #79313 #79346 #79362 #79368 #79386 #79409 #79427 #79428 #79433 #79444
4f275c3e07 Merge #77852 #77949 #79354
84e11ae49b Merge #79186
a3958c6ffb Merge #79049 #79337
b7b37f417c Merge #79359
3d6f41bb04 Merge #79133
b89a636920 Merge #79406
8ac1611f91 Merge #78664 #79272
63ea9139e2 Merge #78983
5d5d0c994e Merge #77847 #79021 #79258
22dc9e987f Merge #79330 <-- supposedly good

**Comment by tbg** [comment]
@renatolabs I see that you've gotten some of the numbers out, thanks for that! Are you pushing through to the end? Looking at what you have so far, I would guess that https://github.com/cockroachdb/cockroach/pull/79454 is the bad commit (just because nothing else looks likely to cause a regression), but it would be good to confirm that. You could, for example, run https://github.com/cockroachdb/cockroach/commit/b84398d95ae301eda8b250a108ab0548ef7f2bf6 with that PR reverted to get, hopefully, a green result.

**Comment by tbg** [comment]
Actually looking at the PR description, it's less clear. Maybe we really need to bisect all the way through.

**Comment by renatolabs** [comment]
I'm running the tests on 6cf835c right now; depending on the result of that, I could bisect b84398d further.

**Comment by renatolabs** [comment]
Initially I thought https://github.com/cockroachdb/cockroach/commit/b84398d95ae301eda8b250a108ab0548ef7f2bf6 was a bad commit (avg throughput at 172k), but I reran the test (`ycsb/C/nodes=3/cpu=3`) in the same commit and got a 178k average, a "good commit" result. The results are not as deterministic as I hoped.

**Comment by erikgrinaker** [comment]
Yeah, this one's been really annoying to track down -- I wasn't able to get a good signal either.

It doesn't show up in GCE benchmarks, so there's a possibility it's due to changes in cloud infra too. I think storage mentioned seeing something like that happen to their benchmarks recently as well.

Do we see a clear perf regression across the timespan at all? Might be worth comparing CPU profiles before/after, in that case. Otherwise, I'm inclined to cut our losses here, at least for now, and do a more holistic comparison of 22.1 vs 22.2 as we near the release.

**Comment by tbg** [comment]
@renatolabs how many instances do you run per SHA? I usually do 5 and pick the median result, but often that's overkill. Maybe here, it isn't.
I agree with Erik re: cutting our losses.

**Comment by renatolabs** [comment]
> @renatolabs how many instances do you run per SHA? I usually do 5 and pick the median result, but often that's overkill. Maybe here, it isn't.

The results I've been posting in the bisection list is the average and standard dev. across 5 runs. Most times the variance is relatively low, but sometimes it's also quite large -- for instance, the latest, "supposedly bad" commit, has an average of 174k qps, but the totals go from 170k (which we would judge "bad") to 178k (which we would judge "good").

At this point I'm also leaning towards cutting our losses on this one.
